internal/network/resource_network_integration.go

```diff
@@ -0,0 +1,280 @@
+package network
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/mapdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	incus "github.com/lxc/incus/v6/client"
+	"github.com/lxc/incus/v6/shared/api"
+	"github.com/lxc/terraform-provider-incus/internal/common"
+	"github.com/lxc/terraform-provider-incus/internal/errors"
+	provider_config "github.com/lxc/terraform-provider-incus/internal/provider-config"
+)
+
+// NetworkIntegrationModel resource data model that matches the schema.
+type NetworkIntegrationModel struct {
+	Name        types.String `tfsdk:"name"`
+	Type        types.String `tfsdk:"type"`
+	Description types.String `tfsdk:"description"`
+	Project     types.String `tfsdk:"project"`
+	Remote      types.String `tfsdk:"remote"`
+	Config      types.Map    `tfsdk:"config"`
+}
+
+// NetworkIntegrationResource represent network integration resource.
+type NetworkIntegrationResource struct {
+	provider *provider_config.IncusProviderConfig
+}
+
+// NewNetworkIntegrationResource returns a new network integration resource.
+func NewNetworkIntegrationResource() resource.Resource {
+	return &NetworkIntegrationResource{}
+}
+
+// Metadata for NetworkIntegrationResource.
+func (r NetworkIntegrationResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = fmt.Sprintf("%s_network_integration", req.ProviderTypeName)
+}
+
+// Schema for NetworkIntegrationResource.
+func (r NetworkIntegrationResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
```

Comment:
I think, we need an additional check here to make sure, that the user does not provide the empty string `""` as name, since this causes an error on Incus side while Terraform does accept it `"" != null`.

```suggestion
				},
				Validators: []validator.String{
					stringvalidator.LengthAtLeast(1),
				},
```

@maveonair, @stgraber this is maybe a general question, since I see this to be missing also for other resources on the `name` attribute.

Reply:
Hmm, yeah, empty names are usually not allowed in the API.

The one exception there being for instances where an empty name is allowed but leads to Incus generating a random instance name for you, which may by itself cause issues with Terraform.

I think it'd make sense for all name properties to be required to be at least 1 character long, then leave the more detailed validation after that to Incus.